File: 5.StructualPattern - Decorator/C++/window/window/DecoratedWindowTest.cpp
#include <iostream>

#include "SimpleWindow.h"
#include "HSBDecorator.h"
#include "VSBDecorator.h"

int main() {
		
		// Create a decorated Window with horizontal and vertical scrollbars
		Window *window = new HSBDecorator(new VSBDecorator(new SimpleWindow()));
		
		// Print the Window's description
		cout << window->getDescription() << endl;

		return 0;
}

File: 5.StructualPattern - Decorator/JAVA/src/window/DecoratedWindowTest.java
package window;

public class DecoratedWindowTest {
	public static void main(String[] args) {
		
		// Create a decorated Window with horizontal and vertical scrollbars
		Window window = new HSBDecorator(new VSBDecorator(new SimpleWindow()));
		
		// Print the Window's description
		System.out.println(window.getDescription());
		}
}
